app/controllers/iiif_controller.rb

```diff
@@ -98,71 +102,21 @@ def current_image
   end
 
   def load_image
-    @image ||= StacksImage.new(stacks_image_params.merge(current_ability: current_ability))
+    @image ||= begin
+                 img = model.new(stacks_image_params)
+                 can?(:download, img) ? img : img.restricted
+               end
   end
 
   def image_info
-    info = current_image.info do |md|
-      if can? :download, current_image
-        md.tile_width = 1024
-        md.tile_height = 1024
-      else
-        md.tile_width = 256
-        md.tile_height = 256
-      end
-    end
-
-    info['profile'] =
-      if can? :download, current_image
-        'http://iiif.io/api/image/2/level1'
-      else
-        ['http://iiif.io/api/image/2/level1', { 'maxWidth' => 400 }]
-      end
-
+    info = current_image.info
+    info['profile'] = current_image.profile
     info['sizes'] = [{ width: 400, height: 400 }] unless current_image.maybe_downloadable?
```

Comment:
I don't remember the `maybe_downloadable?` logic, but I think this is also a candidate for pushing into `RestrictedImage`?